File: datastruct/set/AVLSet.java
public class AVLSet<K extends Comparable<K>> implements Set<K> {
    private AVL<K, Object> avl;

    public AVLSet(){
        avl = new AVL<>();
    }

    @Override
    public int getSize() {
        return avl.getSize();
    }

    @Override
    public boolean isEmpty() {
        return avl.isEmpty();
    }

    @Override
    public void add(K k) {
        if(avl.get(k) == null)
            avl.add(k, null);
    }

    @Override
    public boolean contains(K k) {
        return avl.contain(k);
    }

    @Override
    public void remove(K k) {
        avl.remove(k);
    }
}

File: datastruct/set/AVLSet.py
from Set import Set
from AVL import AVL

class AVLSet(Set):

    def __init__(self):
        self.__avl = AVL()

    def is_empty(self):
        return self.__avl.is_empty()
    
    def get_size(self):
        return self.__avl.get_size()

    def add(self, ele):
        self.__avl.add(ele, None)
    
    def remove(self, ele):
        self.__avl.remove(ele)

    def contains(self, ele):
        return self.__avl.contains(ele)